### PR TITLE
Clean up, fixed issue check recursive filetime (#17).

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

File: src/AssetManager.php [hunk 1]
```diff
@@ -330,19 +304,19 @@ private function convertJs(AssetBundle $bundle): AssetBundle
...
-    private function registerAssetBundle(string $name, ?int $position = null): AssetBundle
+    private function registerAssetBundle(string $name, int $position = null): AssetBundle
     {
         if (!isset($this->assetBundles[$name])) {
             $bundle = $this->getBundle($name);
 
             $this->assetBundles[$name] = false;
 
-            // register dependencies
+            /** register dependencies */
```
Comment (samdark) [inline src/AssetManager.php:319]:
```suggestion
            // register dependencies
```
It's not phpdoc, just inline comment.

File: src/AssetManager.php [hunk 2]
```diff
@@ -330,19 +304,19 @@ private function convertJs(AssetBundle $bundle): AssetBundle
      *
      * {@see registerJsFile()} for more details on javascript position.
      *
-     * @return AssetBundle the registered asset bundle instance
-     * @throws InvalidConfigException
+     * @throws InvalidConfigException|RuntimeException if the asset bundle does not exist or a circular dependency
```
Comment (samdark) [inline src/AssetManager.php:307]: Isn't it always throwing `RuntimeException`?

File: src/AssetManager.php [hunk 3]
```diff
@@ -362,13 +336,13 @@ private function registerAssetBundle(string $name, ?int $position = null): Asset
...
             } elseif ($pos > $position) {
-                throw new \RuntimeException(
+                throw new RuntimeException(
                     "An asset bundle that depends on '$name' has a higher javascript file " .
                     "position configured than '$name'."
                 );
             }
 
-            // update position for all dependencies
+            /** update position for all dependencies */
```
Comment (samdark) [inline src/AssetManager.php:345]: It is inline comment and for inline comments there's no need to use phpdoc.

File: src/AssetPublisher.php [hunk 4]
```diff
@@ -557,10 +549,10 @@ private function publishDirectory(string $src, array $options): array
 
         if ($this->linkAssets) {
             if (!is_dir($dstDir)) {
-                FileHelper::createDirectory(\dirname($dstDir), $this->dirMode);
-                try { // fix #6226 symlinking multi threaded
+                FileHelper::createDirectory(dirname($dstDir), $this->dirMode);
+                try { /** fix #6226 symlinking multi threaded */
```
Comment (samdark) [inline src/AssetPublisher.php:553]: It's not a phpdoc. It's bette to use `//` in this case.

File: src/AssetManager.php (hunk 1 above)
Comment (terabytesoftw) [reply src/AssetManager.php:319]: Fixed.

File: src/AssetManager.php (hunk 2 above)
Comment (terabytesoftw) [reply src/AssetManager.php:307]: Fixed.

File: src/AssetManager.php (hunk 3 above)
Comment (terabytesoftw) [reply src/AssetManager.php:345]: Fixed.

File: src/AssetPublisher.php (hunk 4 above)
Comment (terabytesoftw) [reply src/AssetPublisher.php:553]: Fixed.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**